dpks/classification.py

```diff
@@ -0,0 +1,95 @@
+from typing import TYPE_CHECKING, Any
+import xgboost
+
+from sklearn.model_selection import cross_val_score
+import numpy as np
+import shap
+from sklearn.feature_selection import RFECV
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+
+if TYPE_CHECKING:
+    from .quant_matrix import QuantMatrix
+else:
+    QuantMatrix = Any
+
+
+class Classifier(BaseEstimator, ClassifierMixin):
+    X: np.array
+    y: np.array
+    shap_values: list
+    shap_algorithm: str
+    mean_importance: list
+
+    def __init__(
+        self,
+        classifier,
+        shap_algorithm: str = "auto",
+    ):
+        if isinstance(classifier, str):
+            if classifier == "xgboost":
+                self.classifier = xgboost.XGBClassifier(max_depth=30)
+        else:
+            fit_method = getattr(classifier, "fit", None)
+            predict_method = getattr(classifier, "predict", None)
+            if callable(fit_method) and callable(predict_method):
+                self.classifier = classifier
+            else:
+                raise ValueError(
+                    "The classifier does not have a fit and/or predict method"
+                )
+        self.shap_algorithm = shap_algorithm
+
+    def fit(self, X, y):
+        self.classifier.fit(X, y)
+        self.interpret(X)
+        return self
+
+    def predict(self, X):
+        return self.classifier.predict(X)
+
+    def cross_validation(self, X, y, k_folds: int = 5):
+        scores = cross_val_score(self.classifier, X, y, cv=k_folds)
+        print(
```

Comment:
This should return ```scores``` instead of print out some of the scores

Reply:
Fixed